Attempt at supporting secrets

abstrctn opened the pull request:
See #16

oliver006 commented on main_test.go:
```diff
@@ -119,6 +120,25 @@ func TestParseAndRunConfig(t *testing.T) {
+		// Test exposing secret as file
+		{
+			cfgExpectedOk:        true,
+			env:                  map[string]string{"PLUGIN_ACTION": "deploy", "PLUGIN_TOKEN": validGCPKey, "PLUGIN_SECRETS": `{"var_1":"/mnt/path:secretname"}`, "PLUGIN_SERVICE": "my-service", "PLUGIN_IMAGE": "my-image"},
```
Is this correct?  --> `"var_1":"/mnt/path:secretname"`

Shouldn't this be `"/mnt/path" : "secretname:latest"` ?

abstrctn replied:
You're right, thank you!

oliver006 commented on main.go:
```diff
@@ -178,6 +185,19 @@ func CreateExecutionPlan(cfg *Config) ([]string, error) {
+			}
+
+			// we're using ":||:" as the separator for the args, let's hope no one puts that in an env variable value
+			sep := ":||:"
```
nit: you could move this separator out in a const to the top of the file and use it in both places.

oliver006 reviewed (commented):
I think this makes sense and looks good, just two minor comments.
Could you add something to the README on how to use this? Maybe in a separate block as this is in preview only.
And also add a note that this won't work with Drone 0.8 and/or that Drone 0.8 is no longer supported.

abstrctn commented:
Thanks for the feedback @oliver006, and sorry it took me so long to get to it!

I think I've addressed your notes, and added a section to the README about deprecating support for Drone v0.8. (I suppose theoretically it could still work with older versions as long as you don't set any secrets - although it seems unlikely the step would have permission to deploy without any credentials...)

oliver006 commented:
Released as https://github.com/oliver006/drone-cloud-run/releases/tag/v0.4.0
